codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/WebSocketTransport.java

```diff
@@ -0,0 +1,232 @@
+                        if (future.isSuccess()) {
+                            ctx.pipeline().remove(SockJsHandler.class);
+                            ctx.pipeline().remove(CorsHandler.class);
+                            ctx.pipeline().addLast(new WebSocketSendHandler());
+                        } else {
+                            logger.error("Handshake error", future.cause());
+                        }
+                    }
+                });
+                if (ctx.pipeline().get(SessionHandler.class) != null) {
+                    ctx.fireChannelRead(ReferenceCountUtil.retain(req));
+                }
+            }
+        } finally {
+            req.release();
+        }
+    }
+
+    private static String webSocketLocation(final boolean tls, final FullHttpRequest req) {
+        final String protocol = tls ? "wss://" : "ws://";
+        return protocol + req.headers().get(HOST) + req.uri();
+    }
+
+    private static String webSocketLocation(final boolean tls, final FullHttpRequest req, final String path) {
+        final String protocol = tls ? "wss://" : "ws://";
+        return protocol + req.headers().get(HOST) + path;
+    }
+
+    private void handleWebSocketFrame(final ChannelHandlerContext ctx, final WebSocketFrame wsFrame) throws Exception {
+        try {
+            if (wsFrame instanceof CloseWebSocketFrame) {
+                wsFrame.retain();
+                logger.debug("CloseWebSocketFrame received");
+                ctx.fireUserEventTriggered(Event.CLOSE_SESSION);
+                handshaker.close(ctx.channel(), (CloseWebSocketFrame) wsFrame);
+                return;
+            }
+            if (wsFrame instanceof PingWebSocketFrame) {
+                logger.debug("PingWebSocketFrame received");
+                wsFrame.content().retain();
+                ctx.channel().writeAndFlush(new PongWebSocketFrame(wsFrame.content()));
+                return;
+            }
+            if (!(wsFrame instanceof TextWebSocketFrame)) {
+                throw new UnsupportedOperationException(String.format("%s frame types not supported",
+                        wsFrame.getClass().getName()));
+            }
+            final String[] messages = JsonConverter.decode((TextWebSocketFrame) wsFrame);
+            for (String message : messages) {
+                ctx.fireChannelRead(message);
+            }
+        } finally {
+            wsFrame.release();
+        }
+    }
+
+    @Override
+    public void exceptionCaught(final ChannelHandlerContext ctx, final Throwable cause) throws Exception {
+        if (cause instanceof JsonParseException) {
+            logger.trace("Failed to part JSON", cause);
+            ctx.fireUserEventTriggered(Event.CLOSE_SESSION);
+            ctx.channel().writeAndFlush(new CloseWebSocketFrame(1000, "Broken framing"))
+                    .addListener(ChannelFutureListener.CLOSE);
+            //ctx.channel().close();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)